apps/api-journeys/src/app/modules/event/video/video.resolver.ts

```
@@ -19,123 +20,336 @@ import {
   VideoProgressEventCreateInput
 } from '../../../__generated__/graphql'
 import { EventService } from '../event.service'
+import { BlockService } from '../../block/block.service'
+import { VisitorService } from '../../visitor/visitor.service'
 
 @Resolver('VideoStartEvent')
 @UseGuards(GqlAuthGuard)
 export class VideoStartEventResolver {
-  constructor(private readonly eventService: EventService) {}
+  constructor(
+    private readonly eventService: EventService,
+    private readonly blockService: BlockService,
+    private readonly visitorService: VisitorService
+  ) {}
+
   @Mutation()
   async videoStartEventCreate(
     @CurrentUserId() userId: string,
     @Args('input') input: VideoStartEventCreateInput
   ): Promise<VideoStartEvent> {
+    const block: { journeyId: string } = await this.blockService.get(
+      input.blockId
+    )
+    const journeyId = block.journeyId
```

need to throw a UserInput Error here if no block.